### ISSUE TITLE
why it is "for" here? 

### ISSUE THREAD

**Issue by navyxliu** [aws/amazon-ecs-cli#612]
https://github.com/aws/amazon-ecs-cli/blob/33b5630f46b582752332d1a166ff82b9375404e5/ecs-cli/modules/cli/logs/logs_app.go#L142

I am using --follow. My concern is stack overflow. 
I thin it should be "if-statement"

**Comment by PettitWesley** [comment]
@navyxliu have you encountered stack overflow or some other error while using `--follow`? If so, can you please post the error message you saw?

Here is how that code looks:

```
	for context.Bool(flags.FollowLogsFlag) {
		time.Sleep(followLogsWaitTime * time.Second)
		if lastEvent != nil {
			input.SetStartTime(aws.Int64Value(lastEvent.Timestamp) + 1)
		}
		printLogEvents(context, input, cwLogsClient)
	}
```

The way this works is that if the `--follow` flag is present then `context.Bool(flags.FollowLogsFlag)` will be true. So this loop simplifies too:

```
	for true {
		time.Sleep(followLogsWaitTime * time.Second)
		if lastEvent != nil {
			input.SetStartTime(aws.Int64Value(lastEvent.Timestamp) + 1)
		}
		printLogEvents(context, input, cwLogsClient)
	}
```

This loop will run forever, and sleep, then print the logs. This is what we want- this is how `--follow` is actually implemented- it periodically polls for new logs.

**Comment by navyxliu** [comment]
first of all,  I didn't meet stackoverflow for this.  My "ecs-cli logs" quitted silently. it's not this code snippet. 

secondly, I just want to point out that printLogEvents is self-recursive.  it's not necessary to have a dead-loop 'for true' to make it run forever. `if true` is sufficient. it's even not a tail-recursion.

**Comment by PettitWesley** [comment]
@navyxliu oh I see what you mean now.

You are correct, this code doesn't really make any sense. I'm the one who wrote this code; I'm not quire sure what I was thinking when I wrote it- there's recursion here without any base case. I'm going to label this as a bug- if you ran `--follow` long enough then it could eventually cause problems. One interesting thing I found from doing some cursory reading, [goroutines apparently have an infinite stack](https://dave.cheney.net/2013/06/02/why-is-a-goroutines-stack-infinite)- but if the stack grows large enough it can use up too much memory on the system. Luckily this code here is easy to fix.

Thank you for bringing this to our attention! :)
